contracts/crowdsale/price/CapStagedCrowdsale.sol

```diff
@@ -0,0 +1,96 @@
+pragma solidity ^ 0.4.24;
+
+import "../Crowdsale.sol";
+import "../../math/SafeMath.sol";
+import "../../ownership/Ownable.sol";
+
+
+/**
+  * @title CapStagedCrowdsale
+  * @dev Extension of Crowdsale contract that changes the price of tokens regarding of raised ETH. Each stage of crowdsale is defined in StageLimit (cap limit for the stage) and stageRate (stage rate)
+  * Stages must be inserted from first to last with increasing stageLimit.
+  */
+contract CapStagedCrowdsale is Crowdsale, Ownable {
+  using SafeMath for uint256;
+  using SafeMath for uint;
+  bool isActive = false;
+
+  struct Stage {
+    uint256 stageLimit;
+    uint256 stageRate;
+  }
+
+  Stage[] public stages;
+
+  /**
+   * @dev Reverts if crowdsale started - weiRaised is more than 0 or isActive is set to 1 with setCrowdsaleState function
+   */
+  modifier crowdsaleStarted() {
+    require((weiRaised == 0) && (isActive == false));
+    _;
+  }
+
+  /**
+   * @dev Constructor, takes iniital rate, wallet for funds, token address, array od stage limits in wei and stage rates
```

Comment:
array _of_ stage limits

Reply:
Anyway, these are already described below, so I think we can drop this whole line.

Reply:
i've dropped the whole line